logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -244,18 +244,53 @@ def start_workers
 
       filter_queue_client.set_batch_dimensions(batch_size, batch_delay)
 
+      # First launch WorkerLoop initialization in separate threads which concurrently
+      # compiles and initializes the worker pipelines
+
+      worker_init_thread = []
+      worker_loops = Queue.new
+
+      pipeline_workers.times do |t|
+        thread = Thread.new(
+          lir_execution,
```

Comment:
is there a reason to do this in the main thread? it seems a fair bit of work is done in `JavaBasePipelineExt#lirExecution` to convert java objects to usable ruby objects. Additionally, does each thread need its own copy?

Reply:
answered in main comments

Reply:
reworked it per discussion.